lib/WUI/nhttp/send_json.cpp

```diff
@@ -0,0 +1,128 @@
+#include "send_json.h"
+#include "handler.h"
+#include "headers.h"
+
+#include <timing.h>
+#include <segmented_json_macros.h>
+#include <http/chunked.h>
+#include <transfers/monitor.hpp>
+
+using transfers::Monitor;
+using namespace json;
+using namespace http;
+
+namespace nhttp::handler {
+
+json::JsonResult EmptyRenderer::renderState(size_t resume_point, json::JsonOutput &output, Empty &state) const {
+    return generator(resume_point, output);
+}
+
+JsonResult TransferRenderer::renderState(size_t resume_point, json::JsonOutput &output, TransferState &state) const {
+    // Note: We are guaranteed to get a status, because we checked if it is present
+    // before calling this and we set allow_stale to true
+    auto transfer_status = Monitor::instance.status(true);
```

Comment:
I'd say we may either change the comment above to assert (to verify that our expectations are true) or, as you also do the `_G` stuff and `.reset()` below, simply guard the `transfer_status->id` with a check.